pkg/istio-agent/sds-agent.go

```diff
@@ -175,11 +176,15 @@ func DetectSDS(discAddr string, tlsRequired bool) *AgentConf {
 	hostSDSUDS := sdsUdsPathVar.Get()
 	if strings.HasPrefix(hostSDSUDS, "unix:") {
 		// Skip the prefix
-		hostSDSUDS = sdsUdsPathVar.Get()[5:]
-		if _, err := os.Stat(hostSDSUDS); err == nil {
+		if _, err := os.Stat(hostSDSUDS[5:]); err == nil {
 			ac.SDSAddress = hostSDSUDS
+			ac.HasCitadelAgent = true
 		}
 	}
+	if ac.SDSAddress == "" {
+		ac.SDSAddress = "unix:" + LocalSDS
```

Comment:
At line 180, we trim the prefix "unix:" so that ac.SDSAddress have "unix:", but here we set ac.SDSAddress with prefix "unix:". Could you explain the reason of adding this prefix?

Reply:
Envoy config expects unix:
The LocalSDS var is the file we check - no point in splitting it.

ac.SDSAddress currently has unix: for backward compat - that's how the current injector works.
The code is expected to work with the old injector so tests can pass...